src/Classes/KeyGenerator.php

```diff
@@ -49,6 +49,37 @@ public function generateRandom(): string
         return $key;
     }
 
+    /**
+     * Generate a key for the short URL. This method allows you to pass a
+     * seed value to the key generator. If no seed is passed, a random
+     * key will be generated.
+     *
+     * @param string|int|null $seed
+     * @return string
+     */
+    public function generateKeyUsing($seed = null): string
+    {
+        return match (true) {
+            is_string($seed) => $this->generateKeyUsingString($seed),
+            is_numeric($seed) => $this->hashids->encode($seed),
+            default => $this->generateRandom(),
+        };
+    }
+
+    /**
+     * Generate the random key using a string seed. We take characters from
+     * the beginning of the string to avoid creating a large key.
+     *
+     * @param string $seed
+     * @return string
+     */
+    protected function generateKeyUsingString(string $seed): string
+    {
+        return $this->hashids->encodeHex(
```

Comment:
Is this the best approach? I want to generate a key using a string seed. The `hashids` object only accepts hex values in the `encodeHex` method.

Is there another way to approach this problem? I don't like that I'm changing the dev's input and losing some of the randomness by limiting the string to the X first characters.

Reply:
unfortunately the substr isn't a good idea, as using a ULID would break this behaviour as it will only use the first few characters (which in a ULID are the same)

Reply:
Also, if you try decode, the hextobin function requires the hexadecimal input string to have an even length, which isn't guaranteed if you're truncating data

Reply:
That's a really good point that I hadn't noticed! I'll go back to the drawing board and see if there's any other way I might be able to approach this 🙂

Reply:
Hey @JoshTrebilco! The more I'm thinking about this, the more I'm thinking that the `generateUsing` method should probably only support integers. I feel like I have to do too much manipulation inside the method if a string is passed in and that it can't guarantee a unique key that's a suitable length 😞 

So I'm wondering whether this method should just support integers and it should be left to the developer to decide how to build/generate that integer. For example, if it's a UUID, maybe stripping all the `-` and letters so that it's only left with an integer or something like that?

Do you have any thoughts on this? 🙂

Reply:
@ash-jc-allen that seems like a reasonable compromise 🙂